src/navsat_transform.cpp

```diff
@@ -428,9 +428,14 @@ bool NavSatTransform::toLLCallback(
   }
   tf2::Vector3 point(request->map_point.x, request->map_point.y,
     request->map_point.z);
-  mapToLL(
-    point, response->ll_point.latitude, response->ll_point.longitude,
-    response->ll_point.altitude);
+  try {
+    mapToLL(
+      point, response->ll_point.latitude, response->ll_point.longitude,
+      response->ll_point.altitude);
+  } catch (const GeographicLib::GeographicErr & e) {
+    RCLCPP_ERROR_STREAM(this->get_logger(), e.what());
+    return false;
```

Comment:
I don't think this does anything. This function should be (and is probably casted to a) `void` return type.

We could perhaps set the values to NAN?

Reply:
This is a service callback, so by returning false the caller knows it didn't succeed.

The call to `mapToLL()` can throw an exception which we now catch here.

You can easily try this with a
 `ros2 service call /toLL robot_localization/srv/ToLL "map_point:
  x: 9999999.0
  y: 9999999.0
  z: 0.0"`
  Before the fix this would crash the node with the GeoGraphicLib exception, right now it doesn't crash but prints an error and returns false.

Reply:
Returning NaNs now instead to make clear a function call doesn't return anything useful in that case ;-)